zipkin-example/src/main/java/com/example/helloworld/HelloWorldApplication.java

```diff
@@ -47,30 +46,26 @@ public void initialize(Bootstrap<HelloWorldConfiguration> bootstrap) {
         new ZipkinBundle<HelloWorldConfiguration>(getName()) {
           @Override
           public ZipkinFactory getZipkinFactory(HelloWorldConfiguration configuration) {
-            return configuration.getZipkinFactory();
+            return configuration.getZipkin();
           }
         };
     bootstrap.addBundle(zipkinBundle);
   }
 
   @Override
-  public void run(HelloWorldConfiguration configuration, Environment environment) throws Exception {
+  public void run(HelloWorldConfiguration configuration, Environment environment) {
 
     final Optional<HttpTracing> tracing = zipkinBundle.getHttpTracing();
 
-    final Client client;
+    final JerseyClientBuilder clientBuilder;
     if (tracing.isPresent()) {
-      client =
-          new ZipkinClientBuilder(environment, tracing.get())
-              .build(configuration.getZipkinClient());
+      clientBuilder = new ZipkinClientBuilder(environment, tracing.get());
     } else {
-      final ZipkinClientConfiguration clientConfig = configuration.getZipkinClient();
-      client =
-          new JerseyClientBuilder(environment)
-              .using(clientConfig)
-              .build(clientConfig.getServiceName());
+      clientBuilder = new JerseyClientBuilder(environment);
     }
 
+    final Client client = clientBuilder.using(configuration.getZipkinClient()).build(getName());
```

Comment:
client name is often the same as the app, so better to not use same value as remote service name.